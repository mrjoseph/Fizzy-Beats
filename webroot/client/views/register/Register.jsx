import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AuthService from '../../AuthService/AuthService';
import RegistrationForm from '../../components/form/registrationForm';
import { Title } from './Register.styles';

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      status: null,
    };
    this.Auth = new AuthService();
  }

  componentWillMount() {
    if (this.Auth.loggedIn()) {
      this.props.history.replace('/');
    }
  }

  componentDidUpdate() {
    if (this.Auth.loggedIn()) {
      this.props.history.replace('/');
    }
  }

  render() {
    const { status } = this.state;
    return (
      <div className="container">
        <Title>Register</Title>
        {status && (
        <div className="alert alert-primary" role="alert">
          It looks like you already have a account
        </div>
        )}
        <RegistrationForm />
      </div>
    );
  }
}

Register.propTypes = {
  history: PropTypes.shape([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};

export default Register;
