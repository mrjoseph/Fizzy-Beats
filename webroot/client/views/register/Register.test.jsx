import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Register from './Register';

configure({ adapter: new Adapter() });

describe('Register', () => {
  const spy = jest.spyOn(Register.prototype, 'handleSubmit');
  let component;
  const addUserSpy = jest.fn();
  const props = {
    addUser: addUserSpy,
  };
  beforeEach(() => {
    component = shallow(<Register {...props} />);
  });

  describe('When Submitting the registration form', () => {
    describe('submit', () => {
      it('Should call my handleSubmit', () => {
        const form = component.find('#form');
        const state = {
          username: 'trevor',
          password: 'password',
          email: 'test@test.com',
        };
        component.setState(state);
        form.simulate('submit', { preventDefault() {} });
        expect(spy).toHaveBeenCalled();
        expect(addUserSpy).toHaveBeenCalled();
      });
    });
  });
});
