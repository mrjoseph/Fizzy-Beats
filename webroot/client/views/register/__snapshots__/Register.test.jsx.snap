// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register view When loading check if the user is logged in and Should load the registration form snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="indexstyles__Title-sc-1dnxw1y-0 gCvGgx"
  >
    Register
  </h1>
  <div>
    <form
      className="Registerstyles__Form-e9rrf0-3 ddBFjK"
      id="form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          className="Inputstyles__Label-sc-7bsp2f-0 jzlRQo"
          htmlFor="username"
        >
          <input
            className="input Inputstyles__InputStyled-sc-7bsp2f-2 EmRVR"
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="username"
            type="text"
          />
          <span
            className="Inputstyles__Span-sc-7bsp2f-1 kytcxr"
          >
            username
          </span>
        </label>
      </div>
      <div
        className="field"
      >
        <label
          className="Inputstyles__Label-sc-7bsp2f-0 jzlRQo"
          htmlFor="email"
        >
          <input
            className="input Inputstyles__InputStyled-sc-7bsp2f-2 EmRVR"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="email"
            type="text"
          />
          <span
            className="Inputstyles__Span-sc-7bsp2f-1 kytcxr"
          >
            email
          </span>
        </label>
      </div>
      <div
        className="field"
      >
        <label
          className="Inputstyles__Label-sc-7bsp2f-0 jzlRQo"
          htmlFor="password"
        >
          <input
            className="input Inputstyles__InputStyled-sc-7bsp2f-2 EmRVR"
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="password"
            type="password"
          />
          <span
            className="Inputstyles__Span-sc-7bsp2f-1 kytcxr"
          >
            password
          </span>
        </label>
      </div>
      <button
        className="submit btn btn-secondary"
        disabled={true}
        type="submit"
      >
        Sign up
      </button>
    </form>
  </div>
</div>
`;
